# EDA DataSets (Draft)

In [7]:
import pandas as pd
import openpyxl

# load xlsx, sheet_name = '2022_2023'
players2223 = pd.read_excel('../documents_kickoff_SFC/players_SL_v2_rawdata.xlsx', sheet_name='players_2022_2023')
players2324 = pd.read_excel('../documents_kickoff_SFC/players_SL_v2_rawdata.xlsx', sheet_name='players_2023_2024')

# load csv
pm_2223 = pd.read_csv('../documents_kickoff_SFC/playing_minutes_2022_2023.csv')
pm_2324 = pd.read_csv('../documents_kickoff_SFC/playing_minutes_2023_2024.csv')

## players_SL_v2

### Season 2022/23

In [8]:
players2223.sample(20)

Player  Minutes Played  Starting  Substitute  Substituted In  \
170      Kastriot Imeri            1401        15          14              12   
189         Liam Millar            1448        15          17               9   
261         Noah Streit               0         0           1               0   
196          Luca Zuffi             750         7          25              11   
67        Daouda Guindo            1304        14           2               1   
342       Wylan Cyprien            1370        17           2               1   
337        Víctor Ruiz             161         2           0               0   
54   Carmine Chiappetta             146         1          14               5   
172  Kevin Fickentscher             249         3          29               0   
308   Shkelqim Demhasaj             633         7          16               6   
301    Sayfallah Ltaief             788         7          16              14   
0          Aaron Akalé              10         0           3               1   
93         Eris Abedini            1319        12          23              15   
232     Mirlind Kryeziu            1018        12          16               1   
205      Marc Hornschuh             681         7          26              11   
64      Cédric Zesiger            2728        30           2               1   
104         Fidan Aliti            2363        27           6               2   
129          Hugo Novoa             495         5          14               7   
269     Pascal Schürpf            1913        21          10               9   
78         Diego Heller               0         0           7               0   

     Substituted Out            Position  Age  Value                     Club  \
170               10       Left Midfield   23  6.000           BSC Young Boys   
189                9         Left Winger   23  1.700            FC Basel 1893   
261                0         Left Winger   17  0.025            FC Basel 1893   
196                4    Central Midfield   33  0.100                  FC Sion   
67                 0           Left-Back   20  1.200       FC St. Gallen 1879   
342                8    Central Midfield   28  1.800                  FC Sion   
337                1  Attacking Midfield   29  0.700       FC St. Gallen 1879   
54                 1       Left Midfield   20  0.200            FC Winterthur   
172                1          Goalkeeper   34  0.100                  FC Sion   
308                6      Centre-Forward   27  0.200  Grasshopper Club Zurich   
301                6         Left Winger   23  0.400            FC Winterthur   
0                  0      Centre-Forward   18  0.100            FC Basel 1893   
93                 5  Defensive Midfield   24  0.400            FC Winterthur   
232                2         Centre-Back   26  0.400               FC Zürich   
205                6         Centre-Back   32  0.200               FC Zürich   
64                 1         Centre-Back   25  5.500           BSC Young Boys   
104                6         Centre-Back   29  1.300               FC Zürich   
129                3          Right-Back   20  4.300            FC Basel 1893   
269               18       Left Midfield   33  0.200                FC Luzern   
78                 0          Goalkeeper   18  0.150                FC Luzern   

     Birthdate Height   Foot ContractBegin  ContractEnd  Club_Number  \
170 2000-06-27  1,78m  right    2022-08-16          NaN        452.0   
189 1999-09-27  1,81m  right    2021-07-08          NaN         26.0   
261 2005-10-11  1,79m   left           NaT          NaN         26.0   
196 1990-03-27  1,82m   left    2021-07-01          NaN        321.0   
67  2002-10-14  1,83m   left    2022-07-01          NaN        257.0   
342 1995-01-28  1,80m  right    2022-08-10          NaN        321.0   
337 1993-11-02  1,73m  right    2024-02-14          NaN        257.0   
54  2003-03-09  1,75m  right    2023-

**Insights**:
- `Club_Number`
    - Servette FC = 61
    - BSC Young Boys = 452
    - FC BAsel 1893 = 26

In [5]:
# show all rows with value 'Servette FC' in column 'Club'
players[players['Club'] == 'Servette FC']

Player  Minutes Played  Starting  Substitute  \
14         Alexander Lyng               0         0           8   
16         Alexis Antunes            1508        18           6   
25          Anthony Baron             737         8          11   
47        Bendegúz Bolla            1046        11          14   
57        Bradley Mazikou            1918        23           4   
69            Chris Bedia            1249        14           3   
83          David Douline             827         9           8   
87          Dereck Kutesa            2098        24           5   
101           Dylan Bronn             366         4           3   
110         Enzo Crivelli            1327        13          11   
130          Gaël Ondoua            1872        20           8   
141        Hussayn Touati             146         1          16   
148           Issa Kaloga               0         0           2   
155    Jarell Njiké Simo               0         0           1   
175            Joël Mall            2025        23           4   
184        Jérémy Frick             675         7          17   
185  Jérémy Guillemenot            1126        13          16   
187    Jérôme Onguéné              76         1           5   
192       Keigo Tsunemoto            2036        23           2   
229           Léo Besson               0         0           9   
269   Miroslav Stevanovic            1882        21           2   
312      Patrick Pflücke              58         1           3   
335         Ronny Rodelin              52         0           8   
339            Samba Diba             153         1          18   
363        Steve Rouiller            2494        29           0   
366      Takuma Nishimura             417         5           2   
374          Théo Magnin             235         2          18   
376      Tiemoko Ouattara               0         0          12   
379       Timothé Cognat            2505        29           1   
389         Vasco Tritten               0         0           1   
402          Yoan Severin            2306        25           4   

     Substituted In  Substituted Out            Position  Age  Value  \
14                0                0        Right Winger   19    0.5   
16                4               13  Attacking Midfield   23    2.3   
25                3                2         Centre-Back   31    0.3   
47               11                7      Right Midfield   24    2.1   
57                0                6           Left-Back   27    1.4   
69                3                7      Centre-Forward   27    NaN   
83                5                4  Defensive Midfield   30    0.7   
87                5               15         Left Winger   26    2.0   
101               1                0         Centre-Back   28    1.5   
110              11                8      Centre-Forward   29    0.9   
130               6                3  Defensive Midfield   28    0.9   
141               5                1         Left Winger   22    NaN   
148               0                0          Right-Back   20    0.2   
155               0                0    Central Midfield   19    NaN   
175               0                1          Goalkeeper   33    0.5   
184               1                0          Goalkeeper   31    0.5   
185              13                7      Centre-Forward   26    1.0   
187               1                1         Centre-Back   26    0.5   
192               0                2          Right-Back   25    2.0   
229               0                0          Goalkeeper   21    0.1   
269               1                1        Right Winger   33    0.8   
312               1                1         Left Winger   26    NaN   
335               5                0        Right Winger   33    NaN   
339               8                1    Central Midfield   20    0.5   
363               0                3         Centre-Back   33    0.2   
36

### Season 2023/2024

In [10]:
players2324.sample(20)

Player  Minutes Played  Starting  Substitute  \
327  Richard van der Venne              59         0          10   
235          Markus Kuster             540         6          25   
332            Roman Macek            1370        16          14   
52        Bledian Krasniqi            1388        15          12   
263     Mihailo Stevanovic            1103        10          20   
354          Shaho Maroufi              29         0           2   
366       Takuma Nishimura             417         5           2   
262          Mihael Klepac             158         1          14   
186       Jérémy Vachoux            1170        13           9   
20            Amadou Dante             630         7           1   
240          Marvin Keller            2250        25           5   
403            Yusuf Demir             566         6           8   
326     Riccardo Calafiori             253         3           0   
154          Jaouen Hadjam             566         6           6   
270           Mischa Beeli               0         0           1   
278     Mohsen Abdeljaoued               0         0           2   
39          Aurèle Amenda            1521        17          13   
201          Lars Villiger            1130        12          18   
103          Ebrima Colley             968        11          12   
47         Bendegúz Bolla            1046        11          14   

     Substituted In  Substituted Out            Position  Age  Value  \
327               5                0  Attacking Midfield   31    0.1   
235               0                0          Goalkeeper   30    0.4   
332              11               13    Central Midfield   26    0.8   
52                8                4  Attacking Midfield   22    1.0   
263              15                4    Central Midfield   22    0.8   
354               2                0    Central Midfield   30    NaN   
366               2                5      Second Striker   27    1.2   
262              10                1         Left Winger   26    0.4   
186               0                0          Goalkeeper   29    0.5   
20                1                1           Left-Back   23    2.0   
240               0                0          Goalkeeper   21    1.5   
403               5                6        Right Winger   20    3.0   
326               0                1           Left-Back   21    NaN   
154               3                3           Left-Back   21    1.5   
270               0                0         Centre-Back   20    NaN   
278               0                0      Right Midfield   19    NaN   
39                1                1         Centre-Back   20    8.0   
201              15               12      Centre-Forward   20    1.5   
103               9               10        Right Winger   24    2.6   
47               11                7      Right Midfield   24    2.1   

                        Club  Birthdate Height   Foot ContractBegin  \
327       FC St. Gallen 1879 1992-05-16  1,73m   left   Jul 5, 2023   
235            FC Winterthur 1994-02-22  1,95m  right   Jan 1, 2023   
332                FC Lugano 1997-04-18  1,88m  right  Jan 27, 2019   
52                FC Zürich 2001-06-17  1,73m  right   Jul 3, 2018   
263       FC St. Gallen 1879 2002-01-04  1,86m  right   Jul 1, 2023   
354  FC Stade-Lausanne-Ouchy        NaT    NaN    NaN           NaN   
366              Servette FC 1996-10-22  1,78m  right  Feb 10, 2024   
262         Yverdon Sport FC 1997-09-19  1,84m  right  Jul 18, 2023   
186  FC Stade-Lausanne-Ouchy 1994-07-07  1,83m  right  Jul 26, 2023   
20                FC Zürich 2000-10-07  1,78m   left   Feb 8, 2024   
240            FC Winterthur 2002-07-03  1,89m  right  Sep 13, 2023   
403            FC Basel 1893 2003-06-02  1,71m   left  Aug 16, 2023   
326            FC Basel 1893        NaT    NaN    NaN           NaN   
154           BSC Young Boys 2003-03-26  1,85m   left  Jan 19, 2024   
270       FC 

## playing_minutes_2022_2023